### PR TITLE
JBPM-5976 - Provide auto acknowledge jobs for execution errors

### PR DESCRIPTION
**Pull request by mswiderski**
depends on https://github.com/kiegroup/droolsjbpm-knowledge/pull/242

### CODE REVIEW

File: jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/commands/error/ProcessAutoAckErrorCommand.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    protected List<ExecutionErrorInfo> findErrorsToAck(EntityManager em) {
+        List<ExecutionErrorInfo> errorsToAck = new ArrayList<>();
+        
+        String findProcessErrorsQuery = "select error from ExecutionErrorInfo error "
+                + "where error.acknowledged =:acknowledged "
+                + "and error.processInstanceId in (select pil.id from ProcessInstanceLog pil where status in (:status))";
+        
+        List<ExecutionErrorInfo> processErrorsToAck = em.createQuery(findProcessErrorsQuery)
+                .setParameter("acknowledged", false)
+                .setParameter("status", Arrays.asList(2, 3))
```
Comment (tsurdilo) [inline jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/commands/error/ProcessAutoAckErrorCommand.java]: Could we replace the 2, 3 with ProcessInstance.STATE_COMPLETED, ProcessInstance.STATE_ABORTED?

Comment (mswiderski) [inline jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/commands/error/ProcessAutoAckErrorCommand.java]: That's very good point

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/error/filters/DBExecutionErrorFilter.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ public boolean accept(ExecutionErrorContext errorContext) {
     
     @Override
     public ExecutionError filter(ExecutionErrorContext errorContext) {
-        Builder errorBuilder = ExecutionError.builder().type("DB");
```
Comment (tsurdilo) [inline jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/error/filters/DBExecutionErrorFilter.java]: Would be good to have constants for the builder types, right now they are hard-coded strings like "DB", "Task". WDYT?

Comment (mswiderski) [inline jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/error/filters/DBExecutionErrorFilter.java]: Makes sense, will update

**Comment by mswiderski** [comment]
jenkins retest this

**Comment by mswiderski** [comment]
@tsurdilo there is knowledge PR for this too which I think was not merged

**Comment by tsurdilo** [comment]
@mswiderski yes sorry - merged https://github.com/kiegroup/droolsjbpm-knowledge/commit/c455e7e1eb48c47103ed4b0c9805a66d690c4b1d

let me know if there is anything else i missed please. thanks

**Pull request merged**